azurerm/resource_arm_kubernetes_cluster_test.go

```diff
@@ -548,6 +548,65 @@ func TestAccAzureRMKubernetesCluster_advancedNetworkingAzureNPMPolicyComplete(t
 	})
 }
 
+func TestAccAzureRMKubernetesCluster_autoScalingWithoutAvailabilityZones(t *testing.T) {
+	resourceName := "azurerm_kubernetes_cluster.test"
+	ri := tf.AccRandTimeInt()
+	clientId := os.Getenv("ARM_CLIENT_ID")
+	clientSecret := os.Getenv("ARM_CLIENT_SECRET")
+
+	config := testAccAzureRMKubernetesCluster_autoscaleNoAvailabilityZones(ri, clientId, clientSecret, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMKubernetesClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKubernetesClusterExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "agent_pool_profile.0.min_count", "1"),
+					resource.TestCheckResourceAttr(resourceName, "agent_pool_profile.0.max_count", "2"),
+					resource.TestCheckResourceAttr(resourceName, "agent_pool_profile.0.type", "VirtualMachineScaleSets"),
+					resource.TestCheckResourceAttr(resourceName, "agent_pool_profile.0.enable_auto_scaling", "true"),
+					resource.TestCheckNoResourceAttr(resourceName, "agent_pool_profile.0.availability_zones"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccAzureRMKubernetesCluster_autoScalingWithAvailabilityZones(t *testing.T) {
```

Comment:
Could we get another test that updates these properties?

Reply:
as I mentioned in https://github.com/terraform-providers/terraform-provider-azurerm/pull/317 the moment we change the type of agent pool to VMScaleSets (which is required for autoscaling, but it can be enabled withoutautoscaling). Then any put request to the api will fail (which means no updates will work with that type of agent pool).